### ISSUE TITLE
[FluxStore WooCommerce] red screen NoSuchMethod error on loading

### ISSUE THREAD

**Issue by and-eee** [inspireui/support#3485]
I'm getting this error today on loading the app: 

**NoSuchMethodError: The method [] was called on null. 
Receiver: null
Tried calling []("OnBoarding")
See also:
https://flutter.dev/docs/testing/errors**

![image](https://user-images.githubusercontent.com/63891246/83326155-ecdde700-a2b4-11ea-8b44-f0da126de052.png)

I've tried starting from a fresh download but after reconfiguring everthing I got the same result.

Console log:

`Launching lib/main.dart on iPhone 8 in debug mode...
Running pod install...                                             24.9s
Running Xcode build...
Xcode build done.                                           214.0s
Configuring the default Firebase app...
Configured the default Firebase app __FIRAPP_DEFAULT.
)
)
)
)
Request <OSRequestRegisterUser: 0x60000025e6d0> success result {
    id = "036cf696-0f6f-43c7-9fd9-55453bab269f";
    success = 1;
}
Debug service listening on ws://127.0.0.1:53407/liyb0_u2Tao=/ws
Syncing files to device iPhone 8...
WARNING: OneSignal has detected that your application delegate implements a deprecated method (application:didReceiveLocalNotification:). Please note that this method has been officially deprecated and the OneSignal SDK will no longer call it. You should use UNUserNotificationCenter instead
)
)
)
)
flutter: [2020-05-30 10:45:39.400713Z][Fluxstore][main] ============== main.dart START ==============
ERROR: Error! iOS Simulator does not support push! Please test on a real iOS device. Error: Error Domain=NSCocoaErrorDomain Code=3010 "remote notifications are not supported in the simulator" UserInfo={NSLocalizedDescription=remote notifications are not supported in the simulator}
flutter: [2020-05-30 10:45:39.559382Z][Fluxstore][AppState] initState
flutter: [2020-05-30 10:45:39.567370Z][Fluxstore][AppState] build
flutter: [2020-05-30 10:45:39.594873Z][Fluxstore][AppState] build Theme
flutter: screenName /
flutter: [2020-05-30 10:45:40.592247Z][Fluxstore][AppState] Inital Data
flutter: [2020-05-30 10:45:40.629447Z][Fluxstore][Services] setAppConfig: --> woo <--
flutter: [2020-05-30 10:45:40.780474Z][Fluxstore][AppState] init mobile modules ..
ERROR: Error! iOS Simulator does not support push! Please test on a real iOS device. Error: Error Domain=NSCocoaErrorDomain Code=3010 "remote notifications are not supported in the simulator" UserInfo={NSLocalizedDescription=remote notifications are not supported in the simulator}
flutter: [2020-05-30 10:45:40.836302Z][Fluxstore][AppState] register modules .. DONE
ERROR: Error! iOS Simulator does not support push! Please test on a real iOS device. Error: Error Domain=NSCocoaErrorDomain Code=3010 "remote notifications are not supported in the simulator" UserInfo={NSLocalizedDescription=remote notifications are not supported in the simulator}
flutter: [2020-05-30 10:45:40.884669Z][Fluxstore][AppState] Init Data Finish
flutter: [2020-05-30 10:45:40.931664Z][Fluxstore][Category] getCategories
flutter: url https://app.whitehenry.com.au/wp-json/wc/v2/products/categories?exclude=311&per_page=100&page=1&lang=en
flutter: [2020-05-30 10:45:40.941378Z][Fluxstore][wocommerce_api][20:45:40.940921] getAsync START [endPoint:products/categories?exclude=311&per_page=100&page=1&lang=en] url:https://app.whitehenry.com.au/wp-json/wc/v2/products/categories?exclude=311&per_page=100&page=1&lang=en&consumer_key=ck_5d43771f937646dbd4d6875fa9c9a89ccf325023&consumer_secret=cs_3b43836864ad844335a73a282cc118048d034bb2
flutter: url https://app.whitehenry.com.au/wp-json/wc/v2/products/tags
flutter: [2020-05-30 10:45:40.991511Z][Fluxstore][wocommerce_api][20:45:40.991402] getAsync START [endPoint:products/tags] url:https://app.whitehenry.com.au/wp-json/wc/v2/products/tags?consumer_key=ck_5d43771f937646dbd4d6875fa9c9a89ccf325023&consumer_secret=cs_3b43836864ad844335a73a282cc118048d034bb2
flutter: url https://app.whitehenry.com.au/wp-json/wc/v2/products/attributes
flutter: [2020-05-30 10:45:40.996165Z][Fluxstore][wocommerce_api][20:45:40.996049] getAsync START [endPoint:products/attributes] url:https://app.whitehenry.com.au/wp-json/wc/v2/products/attributes?consumer_key=ck_5d43771f937646dbd4d6875fa9c9a89ccf325023&consumer_secret=cs_3b43836864ad844335a73a282cc118048d034bb2
flutter: [2020-05-30 10:45:41.007144Z][Fluxstore][MyConnectivity] online
flutter: [2020-05-30 10:45:41.014009Z][Fluxstore][App] internet issue change: {ConnectivityResult.wifi: true}
flutter: [2020-05-30 10:45:41.022847Z][Fluxstore][AppState] build Theme
WARNING: OneSignal has detected that your application delegate implements a deprecated method (application:didReceiveLocalNotification:). Please note that this method has been officially deprecated and the OneSignal SDK will no longer call it. You should use UNUserNotificationCenter instead
flutter: screenName null
flutter: ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
flutter: The following NoSuchMethodError was thrown building OnBoardScreen(dirty, state:
flutter: _OnBoardScreenState#21420):
flutter: The method '[]' was called on null.
flutter: Receiver: null
flutter: Tried calling: []("OnBoarding")
flutter:
flutter: The relevant error-causing widget was:
flutter:   OnBoardScreen file:///Users/andy/Development/FluxStore/source/lib/app_init.dart:89:45
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
flutter: #1      _OnBoardScreenState.build (package:fstore/screens/home/onboard_screen.dart:180:40)
flutter: #2      StatefulElement.build (package:flutter/src/widgets/framework.dart:4619:28)
flutter: #3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4502:15)
flutter: #4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4675:11)
flutter: #5      Element.rebuild (package:flutter/src/widgets/framework.dart:4218:5)
flutter: #6      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4481:5)
flutter: #7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:4666:11)
flutter: #8      ComponentElement.mount (package:flutter/src/widgets/framework.dart:4476:5)
flutter: ...     Normal element mounting (24 frames)
flutter: #32     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3446:14)
flutter: #33     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:5947:32)
flutter: ...     Normal element mounting (119 frames)
flutter: #152    Element.inflateWidget (package:flutter/src/widgets/framework.dart:3446:14)
flutter: #153    Element.updateChild (package:flutter/src/widgets/framework.dart:3214:18)
flutter: #154    RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:5580:32)
flutter: #155    MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:5957:17)
flutter: #156    Element.updateChild (package:flutter/src/widgets/framework.dart:3201:15)
flutter: #157    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4527:16)
flutter: #158    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4675:11)
flutter: #159    Element.rebuild (package:flutter/src/widgets/framework.dart:4218:5)
flutter: #160    StatefulElement.update (package:flutter/src/widgets/framework.dart:4707:5)
flutter: #161    Element.updateChild (package:flutter/src/widgets/framework.dart:3201:15)
flutter: #162    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4527:16)
flutter: #163    Element.rebuild (package:flutter/src/widgets/framework.dart:4218:5)
flutter: #164    ProxyElement.update (package:flutter/src/widgets/framework.dart:4862:5)
flutter: #165    _InheritedNotifierElement.update (package:flutter/src/widgets/inherited_notifier.dart:181:11)
flutter: #166    Element.updateChild (package:flutter/src/widgets/framework.dart:3201:15)
flutter: #167    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:5837:14)
flutter: #168    Element.updateChild (package:flutter/src/widgets/framework.dart:3201:15)
flutter: #169    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4527:16)
flutter: #170    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4675:11)
flutter: #171    Element.rebuild (package:flutter/src/widgets/framework.dart:4218:5)
flutter: #172    StatefulElement.update (package:flutter/src/widgets/framework.dart:4707:5)
flutter: #173    Element.updateChild (package:flutter/src/widgets/framework.dart:3201:15)
flutter: #174    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:5837:14)
flutter: #175    Element.updateChild (package:flutter/src/widgets/framework.dart:3201:15)
flutter: #176    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:5837:14)
flutter: #177    Element.updateChild (package:flutter/src/widgets/framework.dart:3201:15)
flutter: #178    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4527:16)
flutter: #179    Element.rebuild (package:flutter/src/widgets/framework.dart:4218:5)
flutter: #180    StatelessElement.update (package:flutter/src/widgets/framework.dart:4583:5)
flutter: #181    Element.updateChild (package:flutter/src/widgets/framework.dart:3201:15)
flutter: #182    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4527:16)
flutter: #183    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4675:11)
flutter: #184    Element.rebuild (package:flutter/src/widgets/framework.dart:4218:5)
flutter: #185    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2627:33)
flutter: #186    WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:883:20)
flutter: #187    RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:284:5)
flutter: #188    SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1113:15)
flutter: #189    SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1052:9)
flutter: #190    SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:968:5)

flutter: #194    _invoke (dart:ui/hooks.dart:261:10)
flutter: #195    _drawFrame (dart:ui/hooks.dart:219:3)
flutter: (elided 3 frames from dart:async)
flutter:
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
[VERBOSE-2:ui_dart_state.cc(157)] Unhandled Exception: Looking up a deactivated widget's ancestor is unsafe.
At this point the state of the widget's element tree is no longer stable.
To safely refer to a widget's ancestor in its dispose() method, save a reference to the ancestor by calling dependOnInheritedWidgetOfExactType() in the widget's didChangeDependencies() method.
#0      Element._debugCheckStateIsActiveForAncestorLookup.<anonymous closure> (package:flutter/src/widgets/framework.dart:3781:9)
#1      Element._debugCheckStateIsActiveForAncestorLookup (package:flutter/src/widgets/framework.dart:3795:6)
#2      Element.findAncestorStateOfType (package:flutter/src/widgets/framework.dart:3914:12)
#3      Navigator.of (package:flutter/src/widgets/navigator.dart:2115:19)
#4      _AnimatedSplashState.build.<anonymous closure> (package:fstore/widgets/common/animated_splash.dart:75:23)
#5      _rootRunUnary (dart:async/zone.dart:1192:38)
#6      _CustomZone.runUnary (dart:async/zone.dart:1085:19)
#7   <…>
flutter: [2020-05-30 10:45:44.092991Z][Fluxstore][wocommerce_api][20:45:44.092819] getAsync END [endPoint:products/attributes] url:products/attributes
flutter: url https://app.whitehenry.com.au/wp-json/wc/v2/products/attributes/28/terms?per_page=100
flutter: [2020-05-30 10:45:44.100263Z][Fluxstore][wocommerce_api][20:45:44.099727] getAsync START [endPoint:products/attributes/28/terms?per_page=100] url:https://app.whitehenry.com.au/wp-json/wc/v2/products/attributes/28/terms?per_page=100&consumer_key=ck_5d43771f937646dbd4d6875fa9c9a89ccf325023&consumer_secret=cs_3b43836864ad844335a73a282cc118048d034bb2
flutter: [2020-05-30 10:45:44.106105Z][Fluxstore][wocommerce_api][20:45:44.105242] getAsync END [endPoint:products/tags] url:products/tags
flutter: [2020-05-30 10:45:44.118136Z][Fluxstore][wocommerce_api][20:45:44.117880] getAsync END [endPoint:products/categories?exclude=311&per_page=100&page=1&lang=en] url:products/categories?exclude=311&per_page=100&page=1&lang=en
flutter: [2020-05-30 10:45:45.732892Z][Fluxstore][wocommerce_api][20:45:45.732741] getAsync END [endPoint:products/attributes/28/terms?per_page=100] url:products/attributes/28/terms?per_page=100
flutter: [2020-05-30 10:45:47.632516Z][Fluxstore][AppState] build
flutter: [2020-05-30 10:45:47.633056Z][Fluxstore][AppState] build Theme

════════ Exception caught by widgets library ═══════════════════════════════════════════════════════
The following NoSuchMethodError was thrown building OnBoardScreen(dirty, state: _OnBoardScreenState#21420):
The method '[]' was called on null.
Receiver: null
Tried calling: []("OnBoarding")

The relevant error-causing widget was: 
  OnBoardScreen file:///Users/andy/Development/FluxStore/source/lib/app_init.dart:89:45
When the exception was thrown, this was the stack: 
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      _OnBoardScreenState.build (package:fstore/screens/home/onboard_screen.dart:180:40)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:4619:28)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4502:15)
#4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4675:11)
...
════════════════════════════════════════════════════════════════════════════════════════════════════
`

I am also getting an *IDE and Plugin Updates: Android Studio is ready to update.* event, I'm not sure if it's related but I have not run the update because this project was working fine yesterday.

Thanks for your assistance guys.

**Comment by minhcasi** [comment]
@and-eee - pls try to compare the default config_en.json from the download package to check if you have update any missing config? Please attach your new common/config folder so that we help for checking

**Comment by and-eee** [comment]
Thank you Minh, in troubleshooting myself I have switched to a very basic output from FluxBuiler to see if it was my own config that was failing but that hasn't helped. 

I've loaded the Config and Common folders to the uploads page, please let me know if there's anything else you need. Cheers.

**Comment by mixkey23** [comment]
Hi, if it contributes something, in my case when I change the DefaultLanguage variable something different that 'en' causes that error.. So I switch back

**Comment by enriqueleon80** [comment]
I've seen this issue before where I had caching turned on but I had an issue on the server serving the JSON file.  Fixed the server and it fixed the app.

**Comment by and-eee** [comment]
Thanks for chiming in @mixkey23 and @enriqueleon80, I haven't changed the language and I did turn caching on to test it. It worked but I quickly turned it back off to continue to work on the app locally. I wonder if that's what caused this? Not sure how to resolve if this is the issue.

**Comment by and-eee** [comment]
Ok have scrapped the project and started fresh (again). Issue is gone, but will update if it starts to happen again.

**Comment by minhcasi** [comment]
@and-eee - please try to make sure this `OnBoarding` from the config json is available for your all config json - this is new config that we add form Fluxbuilder to support the change the Onboarding screen via Fluxbuilder tool 

<img width="1041" alt="Screen Shot 2020-06-01 at 16 40 27" src="https://user-images.githubusercontent.com/1459805/83396715-9a243c80-a426-11ea-9357-f813d26bc253.png">

**Comment by and-eee** [comment]
Hi @minhcasi, it's not there, there is no reference to the onboarding screens in the config_en.json . I didn't even realise Fluxbuilder had the ability to manage the onboarding screens. When I look at the other config_XX.json though I can see it there. So should Fluxbuilder be including those lines? Am I using the wrong version or am I just using it wrong?

**Comment by and-eee** [comment]
Ok, so it has happened again. This is what I did:

1. Ran project in **Android Studio**, linked to site successfully, applied **aconfig_en.json** produced by **FluxBuilder** and tested on **Simulator** successfully.
2. Opened project in **Xcode**, **changed Display Name** in **Xcode** (this may have been a mistake), tested successfully in **Simulator** from **Xcode**.
3. Updated const kSplashScreen = "**assets/images/splashscreen.gif;"** in **images.dart**
4. Opened project in Android Studio again and it wouldn't load the right project (**CONFUSED** but assumed it was to do with the change of name because another project was being loaded instead), So changed it back to see it now go to a white screen after showing the flashscreen which is an . Test **failed**.
5. Went back to **Xcode** and changed **Display Name** back to _Runner_.
6. Went back to **Android Studio** and changed **kSplashScreen** back to the original setting.

So I'm a bit stumped. My days are turning into Groundhog Days and not enjoyig it at all. I hope you geniuses have some good ideas for =me :) I'll be back in 6 hours when I get up to go to work and hopefully won't have to start again for the 4th time, What am I doing wrong??

**Comment by minhcasi** [comment]
@and-eee - pls help to try the app name by editing the `info.plist` file 

<img width="929" alt="Screen Shot 2020-06-02 at 10 59 07" src="https://user-images.githubusercontent.com/1459805/83478468-1880ed00-a4c0-11ea-978f-8cfdbdc73ce1.png">

**Comment by and-eee** [comment]
Hi @minhcasi, while waiting for a response I ran through the whole process again. This time I never got to run the working app at all, it went straight to the red screen. I had been very careful not to do anything in Xcode this time and I went through the Documentation very carefully, including the 'Change the app name' process you mentioned above. The result when running the project in Android Studio is the same red screen, and when running in Xcode I am seeing the flash screen (the Insireui one, unchanged) followed by a white screen.

Is it possible to disable/remove the Onboarding component altogether to see if that's causing an issue or if it is something else?

**Comment by and-eee** [comment]
@minhcasi - Update: I had backed up the config_en.json this time so I restored it and the app is running. I had changed config_en.json to the code from woocommerce example **config-theme1.json** but I had included the onboarding lines you referred to in the screenshot you sent earlier (https://github.com/inspireui/support/issues/3485#issuecomment-636742702).

Having removed those, it is now working. Is it safe to use FluxBuilder to continue development or is it safer to just work through the code manually?
